core/PhysiCell_cell_container.cpp

```diff
@@ -159,54 +161,58 @@ void Cell_Container::update_all_cells(double t, double phenotype_dt_ , double me
 		}
 	}
 	
-	if( time_since_last_cycle > phenotype_dt_ - 0.5 * diffusion_dt_ || !initialzed )
+	if( time_for_phenotype )
 	{
 		// Reset the max_radius in each voxel. It will be filled in set_total_volume
 		// It might be better if we calculate it before mechanics each time 
 		// std::fill(max_cell_interactive_distance_in_voxel.begin(), max_cell_interactive_distance_in_voxel.end(), 0.0);
 		
-		if(!initialzed)
-		{
-			time_since_last_cycle = phenotype_dt_;
-		}
-		
 		// new as of 1.2.1 -- bundles cell phenotype parameter update, volume update, geometry update, 
 		// checking for death, and advancing the cell cycle. Not motility, though. (that's in mechanics)
 		#pragma omp parallel for 
 		for( int i=0; i < (*all_cells).size(); i++ )
 		{
 			if( (*all_cells)[i]->is_out_of_domain == false )
 			{
-				(*all_cells)[i]->advance_bundled_phenotype_functions( time_since_last_cycle ); 
+				(*all_cells)[i]->advance_bundled_phenotype_functions(time_since_last_phenotype);
 			}
 		}
-		
 		// process divides / removes 
 		for( int i=0; i < cells_ready_to_divide.size(); i++ )
 		{
 			cells_ready_to_divide[i]->divide();
 		}
+		// remove cells here (and not only below after interactions) in case phenotype and mechanics time steps don't line up
 		for( int i=0; i < cells_ready_to_die.size(); i++ )
-		{	
-			cells_ready_to_die[i]->die();	
+		{
+			cells_ready_to_die[i]->die();
 		}
 		num_divisions_in_current_step+=  cells_ready_to_divide.size();
 		num_deaths_in_current_step+=  cells_ready_to_die.size();
 		
 		cells_ready_to_die.clear();
 		cells_ready_to_divide.clear();
-		last_cell_cycle_time= t;
-	}
 		
-	double time_since_last_mechanics= t- last_mechanics_time;
-	
-	// if( time_since_last_mechanics>= mechanics_dt || !initialzed)
-	if( time_since_last_mechanics > mechanics_dt_ - 0.5 * diffusion_dt_ || !initialzed )
+		time_since_last_phenotype = diffusion_dt_; // setting it for next cycle
+	}
+	else
+	{ time_since_last_phenotype += diffusion_dt_; }
+
+	if( time_for_mechanics )
 	{
-		if(!initialzed)
+		// new March 2022: 
+		// run standard interactions (phagocytosis, attack, fusion) here 
+		#pragma omp parallel for 
+		for( int i=0; i < (*all_cells).size(); i++ )
+		{
+			Cell* pC = (*all_cells)[i]; 
+			standard_cell_cell_interactions(pC,pC->phenotype,time_since_last_mechanics); 
```

Comment:
We need to discuss this change in call order further, as it will impact phagocytosis, attack, and fusion. I noticed it also relates to other PRs (#340 and #341). This change might introduce too many side effects.

Reply:
Yes, very happy to discuss more. By side effects, do you mean unintended behavior and/or errors? Or do you mean changes from how it ran before?

If we find this to be a long discussion, I would prefer to get a quick fix for this bug while we keep working on a longterm fix. Perhaps something like (the highly inefficient) #339?

Reply:
I mean unintended behavior (like in PR #340 ), as the attach and detach dynamics were working fine in the previous release. I feel that by making this call order change, we might introduce these issues and possibly others we haven’t considered yet. Does that make sense?

Reply:
Yeah, that makes sense. To be clear, #340 and #341 are built off the dev branch / previous release (1.14.0), not this fix. Their point of origin is how we implemented changes in 1.14.0. This fix does not change the existence of those behaviors.